controllers/operandrequest/reconcile_operator.go

```diff
@@ -97,14 +97,19 @@ func (r *Reconciler) reconcileOperator(ctx context.Context, requestInstance *ope
 				// Subscription existing and managed by OperandRequest controller
 				if _, ok := sub.Labels[constant.OpreqLabel]; ok {
 					// Subscription channel changed, update it.
-					if compareSub(sub.Spec, opt) {
+					if compareSub(sub, opt, registryKey) {
 						sub.Spec.CatalogSource = opt.SourceName
 						sub.Spec.Channel = opt.Channel
 						sub.Spec.CatalogSourceNamespace = opt.SourceNamespace
 						sub.Spec.Package = opt.PackageName
 						if opt.InstallPlanApproval != "" && sub.Spec.InstallPlanApproval != opt.InstallPlanApproval {
 							sub.Spec.InstallPlanApproval = opt.InstallPlanApproval
 						}
+						// initialize map interface for upgrade case
+						if sub.Annotations == nil {
+							sub.Annotations = make(map[string]string)
+						}
+						sub.Annotations[registryKey.Namespace+"."+registryKey.Name] = "true"
```

Comment:
@horis233 
PR is updated, for the existing subscription, handle the upgrade scenario and update annotations with other `sepc` field in the same place.